### PR TITLE
Fix FW update flow for BTC-only devices

### PR DESCRIPTION
**Pull request by komret**
## Description

Bitcoin-only device with Universal firmware should offer updates for Universal firmware, not Bitcoin-only, and it should behave the same as a regular device in every way.

Suite (connect) should always recognize current firmware type except in the following cases:
- T1, TT or custom firmware in bootloader mode (in that case, Universal firmware is offered for update)
- TS3 with FW <2.6.3 in bootloader mode (in that case, firmware is offered based on device type (BTC-only or regular)

**QA**: Please test FW install, update and switch on a BTC-only TS3 with 2.6.3 fw (Universal firmware can be installed via custom firmware installation). Would be nice to test with other devices as well.

## Screenshots
Also a minor fix in `FirmwareOffer`, it previously said `Universal` when the user was about to install Bitcoin-only firmware in onboarding.
Screenshot 2023-10-26 at 15 53 32

Screenshot 2023-10-26 at 16 20 50

### CODE REVIEW

**Comment by tomasklim** [comment]
> I also unified some copy, those two versions were nearly identical.

I separated it into two translation strings as both strings are for different cases. The text is almost identical now, but I expected that it should be changed later. The text was not final in the design or the case was not in design at all, i do not remember.

Thats why the description is different.
`Explanation of Bitcoin-only firmware in onboarding` vs `Explanation of Bitcoin-only firmware in onboarding for Bitcoin-only devices`

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 1]
```diff
@@ -160,16 +160,17 @@ export const FirmwareInitial = ({
...
-        (!isCurrentlyBitcoinOnly && !shouldSwitchFirmwareType) ||
-        // attempting to switch to Bitcoin-only from old firmware
-        (!isCurrentlyBitcoinOnly && shouldSwitchFirmwareType && !isBitcoinOnlyAvailable)
-            ? FirmwareType.Regular
-            : FirmwareType.BitcoinOnly;
+        // updating Bitcoin-only
+        (isCurrentlyBitcoinOnly && !shouldSwitchFirmwareType) ||
+        // switching to Bitcoin-only
+        (!isCurrentlyBitcoinOnly && shouldSwitchFirmwareType && isBitcoinOnlyAvailable) ||
+        // installing an empty Bitcoin-only device
```
Comment (tomasklim) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx]:
```suggestion
        // initialising an empty Bitcoin-only device
```

or installing fw on an empty ...

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 2]
```diff
@@ -101,7 +101,7 @@ const getNoFirmwareInstalledSubheading = (device: AcquiredDevice) => {
     if (bitcoinOnlyDevice) {
         return device.firmware === 'none'
             ? 'TR_FIRMWARE_SUBHEADING_NONE_BITCOIN_ONLY_DEVICE'
-            : 'TR_FIRMWARE_SUBHEADING_UNKNOWN_BITCOIN_ONLY_DEVICE';
```
Comment (tomasklim) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx]: I would not do that, see comment above

**Review by tomasklim — DISMISSED** [review]
I like the simplification of the condition. Thank you

File: packages/suite/src/components/firmware/FirmwareInitial.tsx (hunk 2 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: Ok, dropped.

File: packages/suite/src/components/firmware/FirmwareInitial.tsx (hunk 1 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: Simplified further.

File: packages/connect/src/device/Device.ts [hunk 3]
```diff
@@ -544,8 +544,13 @@ export class Device extends TypedEmitter<DeviceEvents> {
         this.features = feat;
         this.featuresNeedsReload = false;
 
-        // Firmware type is unknown in bootloader mode
-        if (this.getMode() !== 'bootloader') {
+        // Vendor headers have been changed in 2.6.3.
+        if (feat.fw_vendor === 'Trezor Bitcoin-only') {
```
Comment (matejkriz) [inline packages/connect/src/device/Device.ts:548]: I'd be inclined to define those possible headers into constants somewhere, but as long as it's a single use, it's probably OK to have strings.

**Review by matejkriz — APPROVED** [review]
nice 👍

**Comment by bosomt** [comment]
**QA OK**

tested multiple/all device models and FW types combinations including BTC only device.

**Pull request merged**